### PR TITLE
dont pad 0 dim mm inputs

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126475
* #125780
* #125773
* #125772

Otherwise you get an error in constant_pad_nd.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by huydhn — APPROVED** [review]

File: torch/_inductor/fx_passes/pad_mm.py [hunk 1]
```diff
@@ -325,6 +326,17 @@ def should_pad_bench(
         if m_padded_length == k_padded_length == n_padded_length == 0:
```
Comment (peterbell10) [inline torch/_inductor/fx_passes/pad_mm.py:326]: Should this condition be `any(x == 0 for x in (...))`.

Comment (eellison) [reply torch/_inductor/fx_passes/pad_mm.py:326]: hmm, not sure I follow. what do you mean exactly ?

Comment (peterbell10) [reply torch/_inductor/fx_passes/pad_mm.py:326]: The quoted code checks if all of m, n and k are 0. Your new code checks if any of the shape variables are 0. 

I think we could cover all cases with one check that any of m, n or k are zero.

Comment (eellison) [reply torch/_inductor/fx_passes/pad_mm.py:326]: the quoted code is for the padded length difference, as opposed to the shapes themselves. the padded length we want to skip if they're all zero, whereas for the shapes themselves we want to skip if any are 0

Comment (peterbell10) [reply torch/_inductor/fx_passes/pad_mm.py:326]: Oh I see, "padded length" actually means "length of padding" here. Confusing name, but preexisting so okay.

Comment (eellison) [reply torch/_inductor/fx_passes/pad_mm.py:326]: agreed..

**Pull request closed without merging**